app/controllers/manageiq/graphql/application_controller.rb

```diff
@@ -1,6 +1,43 @@
 module ManageIQ
   module GraphQL
     class ApplicationController < ActionController::API
+      include ActionController::HttpAuthentication::Basic::ControllerMethods
+
+      def authenticate!
+        return if try_authenticate_with_http_basic || try_authenticate_with_http_token
+        headers["WWW-Authenticate"] = 'Basic realm="Application"'
+        render :status => 401, :json => { :data => nil, :errors => [ { 'message' => 'Unauthorized' } ] }
+      end
+
```

Comment:
`private` for the two methods below?

(yeah I know it's sort of moot when you subclass controllers but more about semantics and implying this isn't supposed to be public API)